### PR TITLE
Update: Add more tests for FEN validation

### PR DESCRIPTION
**Pull request by Prashanth-Chandra**
Current tests include 

- Number of Kings should be equal to 1
- Number of pieces must not exceed the maximum pieces
- Number of rows and columns must be 8
- Pawn positions validation
- FEN should only contain valid characters ('k', 'q', 'r', 'n', 'b', 'p', 'K', 'Q', 'R', 'N', 'B', 'P', '/', '1', '2', '3', '4', '5', '6', '7', '8')

Added more clear error messages while validating
Example : if the input is r1bk3r/p2pBpNp/n4n2/1p1NP2P/6P1/3P4/P1P1K3/q5b (The last row q+5+b => 7)
image

### CODE REVIEW

**Comment by Prashanth-Chandra** [comment]
@SrNightmare09 Can you review this PR

File: src/checkFen.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ function checkFen(fenString) {
             }
         }
         if (rowCount != 8) {
-            customAlert("Invalid FEN position\nNumber of columns is not 8 in row " + (i + 1));
+            customAlert("Invalid FEN position\nNumber of columns in row " + (i + 1) + " is not 8");
```
Comment (SrNightmare09) [inline src/checkFen.js:75]: Please change this to number of squares instead of number of columns

File: src/checkFen.js [hunk 2]
```diff
@@ -2,24 +2,56 @@ function checkFen(fenString) {
...
     //check if fenString is valid with respect to the number of pieces
     fenArray = getCapturedPieces(1, fenString);
 
+    const piece = ["King", "Queen", "Rook", "Knight", "Bishop", "Pawn"];
+
     for (let i = 0; i < fenArray.length; i++) {
         if (fenArray[i] < 0) {
             isValid = false;
+            if (i < 6) {
+                customAlert("Invalid FEN position\nNumber of black " + piece[i] + "'s is not valid");
```
Comment (SrNightmare09) [inline src/checkFen.js:19]: Please remove 's as it is grammatically incorrect. Thank you.

Comment (SrNightmare09) [reply src/checkFen.js:19]: For example. 
- number of black rooks is correct
- number of black rook's is incorrect

**Review by SrNightmare09 — COMMENTED** [review]
Great job! I have suggested a few changes in the error messages. Please bring them into effect after which I can merge this PR! Thank you!

**Comment by Prashanth-Chandra** [comment]
@SrNightmare09 Can you review my latest commit

**Comment by SrNightmare09** [comment]
@Prashanth-Chandra thank you for strengthening the FEN validation even more! your efforts are highly appreciated!

**Pull request merged**